smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -139,44 +139,96 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
     }
 
     protected open fun renderAssertions(test: HttpResponseTestCase, outputShape: Shape) {
-        val members = outputShape.members().filterNot { it.hasTrait(HttpQueryTrait::class.java) }
-        val path = ".properties".takeIf { outputShape.hasTrait<ErrorTrait>() } ?: ""
-        renderMemberAssertions(writer, test, members, model, symbolProvider, "expected$path", "actual$path")
+        findShapesRequiringEquatable(outputShape)
+        writer.write("XCTAssertEqual(actual, expected)")
     }
 
-    open class Builder : HttpProtocolUnitTestGenerator.Builder<HttpResponseTestCase>() {
-        override fun build(): HttpProtocolUnitTestGenerator<HttpResponseTestCase> {
-            return HttpProtocolUnitTestResponseGenerator(this)
+    private fun findShapesRequiringEquatable(shape: Shape) {
```

Comment:
Starting with the output shape, traverse the shape's members, recursively looking for shapes that need to be made Equatable.

Keep track of what is already Equatable (in the set `hasBeenRenderedEquatable`) so you don't get multiple conformances.